### PR TITLE
Translate validation error messages in the deserializer

### LINKED ISSUE #1742: Translate validation error messages
Translate validation error messages in the deserializer.

### PR DESCRIPTION
**Pull request by wesleybl**
Fixes #1742

### CODE REVIEW

**Comment by wesleybl** [comment]
@jenkins-plone-org please run jobs

**Review by erral — APPROVED** [review]

**Comment by davisagli** [comment]
@jenkins-plone-org please run jobs

**Review by davisagli — APPROVED** [review]

File: requirements-docs.txt [hunk 1]
```diff
@@ -1,5 +1,10 @@
 docutils<0.17,>=0.15  # sphinx-book-theme 0.2.0 has requirement docutils<0.17,>=0.15
 Sphinx<5,>=3  # sphinx-book-theme 0.3.3 has requirement sphinx<5,>=3
+sphinxcontrib-applehelp==1.0.4  # newer versions require sphinx 5
+sphinxcontrib-devhelp==1.0.2   # newer versions require sphinx 5
+sphinxcontrib-htmlhelp==2.0.1  # newer versions require sphinx 5
+sphinxcontrib-qthelp==1.0.3  # newer versions require sphinx 5
+sphinxcontrib-serializinghtml==1.1.5  # newer versions require sphinx 5
```
Comment (stevepiercy) [inline requirements-docs.txt:7]: Do we really need all those `sphinxcontrib-*help`s?

File: requirements-docs.txt [hunk 2]
```diff
@@ -1,5 +1,10 @@
 docutils<0.17,>=0.15  # sphinx-book-theme 0.2.0 has requirement docutils<0.17,>=0.15
 Sphinx<5,>=3  # sphinx-book-theme 0.3.3 has requirement sphinx<5,>=3
```
Comment (stevepiercy) [inline requirements-docs.txt:2]: Let's unpin this.

```suggestion
Sphinx
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Let's try using the latest version of everything for docs in `requirements.txt`. Untested, but it follows the pattern in https://github.com/plone/training/blob/main/requirements.txt.

File: requirements-docs.txt (hunk 1 above)
Comment (davisagli) [reply requirements-docs.txt:7]: @stevepiercy Yes, they are dependencies of Sphinx. They released new versions yesterday which are not compatible with Sphinx<5, but no longer declare that in their package metadata.

File: requirements-docs.txt (hunk 2 above)
Comment (davisagli) [reply requirements-docs.txt:2]: @stevepiercy Let's sort this out separately. I was just trying to unblock merging the actual change here.

File: requirements-docs.txt (hunk 1 above)
Comment (stevepiercy) [reply requirements-docs.txt:7]: I mean that we don't build those types of outputs, so why add them at all? We use only HTML and linkcheck.

**Comment by stevepiercy** [comment]
Here's a requirements-docs.txt that worked for me.

```
Sphinx
jsx-lexer
lesscpy
linkify-it-py
myst-parser
sphinx-autobuild
sphinx-book-theme
sphinx-copybutton
sphinx-sitemap
sphinx-togglebutton
sphinxcontrib.httpdomain
sphinxcontrib.httpexample
sphinxcontrib-spelling
sphinxext-opengraph
```

Also requires a change to `conf.py`:

```py
    # "extra_navbar": """
    # <p class="ploneorglink">
    #     <a href="https://plone.org">
    #         <img src="/_static/logo.svg" alt="plone.org" /> plone.org</a>
    # </p>""",
```

This should work on RTD where docs get published. It will not affect pulling in docs into the main documentation.

File: requirements-docs.txt (hunk 1 above)
Comment (davisagli) [reply requirements-docs.txt:7]: @stevepiercy And I meant that as far as I can tell, we don't have a choice, because they are pulled in by sphinx.

**Comment by davisagli** [comment]
@stevepiercy I didn't realize that the plone.restapi docs are still published separately at https://plonerestapi.readthedocs.io/

Do we really want that? (Doesn't it hurt SEO of the main docs?)

**Comment by stevepiercy** [comment]
Prior to Plone 6, there was no automatic documentation deploy process, so RTD stood in for it. I can't find the discussion about removing them from RTD. IMO it is a bad idea to publish them to RTD for SEO, branding, Marketing, and additional maintenance. Someone objected to their removal from RTD, the reason for which I cannot recall, and I had no interest in pushing it.

**Comment by davisagli** [comment]
@stevepiercy ok, I found the prior discussion about that: https://github.com/plone/plone.restapi/issues/1447

**Pull request merged**